crates/settlement-clients/starknet/src/lib.rs

```diff
@@ -197,11 +206,26 @@ impl SettlementClient for StarknetSettlementClient {
         if block_number.is_empty() {
             return Err(eyre!("Could not fetch last block number from core contract."));
         }
-        Ok(block_number[0].try_into()?)
+
+        Ok(u64::from_le_bytes(block_number[0].to_bytes_le()[0..8].try_into().unwrap()))
     }
 
     /// Returns the nonce for the wallet in use.
     async fn get_nonce(&self) -> Result<u64> {
         todo!("Yet to impl nonce call for Starknet.")
```

Comment:
can you implement this as well

Reply:
Implemented

Reply:
i would ideally add a small test case for this :)